EasyPost/Models/API/Beta/CarrierMetadata.cs

```diff
@@ -0,0 +1,148 @@
+using System.Collections.Generic;
+using EasyPost.Utilities.Internal;
+using Newtonsoft.Json;
+
+namespace EasyPost.Models.API.Beta
+{
+    public class CarrierMetadata
+    {
+        #region JSON Properties
+
+        [JsonProperty("carriers")]
+        public List<Carrier>? Carriers { get; set; }
+
+        #endregion
+    }
+
+    public class Carrier
+    {
+        #region JSON Properties
+
+        [JsonProperty("name")]
+        public string? Name { get; set; }
+
+        [JsonProperty("human_readable")]
+        public string? HumanReadable { get; set; }
+
+        [JsonProperty("predefined_packages")]
+        public List<PredefinedPackage>? PredefinedPackages { get; set; }
+
+        [JsonProperty("service_levels")]
+        public List<ServiceLevel>? ServiceLevels { get; set; }
+
+        [JsonProperty("shipment_options")]
+        public List<ShipmentOption>? ShipmentOptions { get; set; }
+
+        [JsonProperty("supported_features")]
+        public List<SupportedFeature>? SupportedFeatures { get; set; }
+
+        #endregion
+    }
+
+    public class PredefinedPackage
+    {
+        #region JSON Properties
+
+        [JsonProperty("carrier")]
```

Comment:
nit: the `carrier` and `name` will always be populated on each of these types while everything else is optional. Could remove the question mark